xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp

```diff
@@ -576,6 +576,7 @@ void CVideoPlayerAudio::Process()
         msg.player = VideoPlayer_AUDIO;
         msg.cachetotal = cachetotal;
         msg.cachetime = cachetime;
+        msg.next_pts = m_messageQueue.GetTimeBack();
         msg.timestamp = m_audioClock;
```

Comment:
There is 0.1 sek between the change: Is this OR or is this AND?
Are 0.1 secs enough to handle the Audio stream change?

Is there a video cache similar to the audio cache wich holds decoded video frames waiting for display?
I couldn't believe it because decoder decodes directly into video mem, right?. But then either the PTS's are 0.5 secs unsynchronized befoere the video switch (I cannot believe this too because there was time enough to synchronize) or the buffer state is not correct.

I'll investigate now the PTS at the time of the Audio switch - for me it is really unclear how these 0.5 seconds come from.